web-ide/proxy/src/proxy.ts

```diff
@@ -144,5 +144,7 @@ function addSecureHeaders(res :Response, config :any) {
     res.setHeader("X-Frame-Options", "sameorigin")
     res.setHeader("X-XSS-Protection", "1; mode=block")
     res.setHeader("Referrer-Policy", "no-referrer-when-downgrade")
-    res.setHeader("Content-Security-Policy", `default-src 'self'; connect-src 'self' ws://${config.http.hostname}:${config.http.port}/; script-src 'self' 'unsafe-inline' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; img-src 'self' data:; font-src 'self' data:;`)
+
+    const host = config.http.port === 80 ? config.http.hostname : `${config.http.hostname}:${config.http.port}`
+    res.setHeader("Content-Security-Policy", `default-src 'self'; connect-src 'self' wss://${host}/ ws://${host}/; script-src 'self' 'unsafe-inline' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; img-src 'self' data:; font-src 'self' data:;`)
```

Comment:
Why support insecure websockets at all? Don't we expect to always serve this page under HTTPS, as the HSTS header above suggests?

Also, it may be worth setting up `report-to` and `report-uri` entries, so we can know about these problems before they get reported on Twitter in the future. Ed has set up a server for collecting those reports already, you may want to ask him for the values to set.

Reply:
FYI the value mentioned in [Ed's best practices [internal DA link]](https://digitalasset.atlassian.net/wiki/spaces/SEC/pages/850853931/Web+Site+Web+Services+Security+-+Best+Practices) is:
```report-uri: https://report-uri.digitalasset.com/report-uri```

Reply:
thanks @garyverhaegen-da - I'm adding the report-uri

I'd like to keep the ws:// as well as it allows me to test locally without too much trouble. I simply use http/ws and add the security headers